frontend/src/components/Field/Rating/Rating.tsx

```diff
@@ -108,70 +106,66 @@ export const Rating = forwardRef<RatingProps, 'input'>(
-    )
+    // Generate options from maxRating given, grouped by display row.
+    const options = useMemo(() => {
+      const options = []
+      let suboptions = []
+      for (let i = 1; i <= numberOfRatings; i++) {
+        suboptions.push(i)
+        if (isSplitRows && i % wrapComponentsPerRow === 0) {
+          options.push(suboptions)
+          suboptions = []
+        }
+      }
+      if (suboptions.length > 0) options.push(suboptions)
+      return options
+    }, [isSplitRows, numberOfRatings, wrapComponentsPerRow])
 
     const ratingLayout = useMemo(() => {
       switch (variant) {
         case 'number':
-          return { spacing: '-1px', rowHeight: '0.5rem' }
+          return { spacingX: '-1px', spacingY: '0.5rem' }
         default:
-          return { spacing: 0, rowHeight: 0 }
+          return { spacingX: 0, spacingY: 0 }
       }
     }, [variant])
 
     return (
       <Grid
         rowGap="0.5rem"
-        templateAreas={[
-          `'caption' 'rating'`,
-          `'caption' 'rating'`,
-          `'caption' 'rating'`,
-          `'rating' 'caption'`,
-        ]}
+        templateAreas={{
+          base: "'caption' 'rating'",
+          md: "'rating' 'caption'",
+        }}
       >
         <Stack
           gridArea="rating"
           as="fieldset"
-          direction={['column', 'column', 'column', 'row']}
-          spacing={['0.5rem', '0.5rem', '0.5rem', '1rem']}
-          align={['flex-start', 'flex-start', 'flex-start', 'center']}
+          direction={{ base: 'column', md: 'row' }}
+          spacing={{ base: '0.5rem', md: '1rem' }}
+          align={{ base: 'flex-start', md: 'center' }}
         >
-          <Wrap spacing={ratingLayout.spacing}>
-            {options.map((value, i) => {
-              return (
-                <Fragment key={value}>
-                  <WrapItem>
-                    <RatingOption
-                      name={name}
-                      variant={variant}
-                      colorScheme={colorScheme}
-                      value={value}
-                      onChange={handleRatingChange}
-                      selectedValue={currentValue}
-                      isDisabled={isDisabled}
-                      {...(i === 0 ? { ref } : {})}
-                    >
-                      {value}
-                    </RatingOption>
-                  </WrapItem>
-                  {
-                    // Force component to begin on a new line.
-                    value !== numberOfRatings &&
-                      value % wrapComponentsPerRow === 0 &&
-                      isSplitRows && (
-                        <WrapItem display="contents" aria-hidden>
-                          <Box flexBasis="100%" h={ratingLayout.rowHeight} />
-                        </WrapItem>
-                      )
-                  }
-                </Fragment>
-              )
-            })}
-          </Wrap>
+          <Stack spacing={ratingLayout.spacingY}>
```

Comment:
comment: quite confusing to see spacingY assigned to spacing... should we rename them to something less confusing (left as an exercise to you) since we are not spreading the layout props?

Reply:
[e621225](https://github.com/opengovsg/FormSG/pull/4908/commits/e62122558f4582b9fdf0695a88a417a3224671ab) renamed!